tstate/tstate_view.go

```diff
@@ -222,6 +222,8 @@ func (ts *TStateView) Insert(ctx context.Context, key []byte, value []byte) erro
 	}
 	valueChunks, _ := keys.NumChunks(value) // not possible to fail
 	k := string(key)
+	// Invariant: [getValue] is safe to call here because with [state.Write], it
```

Comment:
I wonder if it makes sense to just explicitly require read for writes?

Reply:
Given that we do so here?

Reply:
On the other hand, it seems like weird DX to have to specify a `Read` for something that doesn't require a `Get` call.

Reply:
I think we need to align this with the cost to the VM, which does indeed need to do a read to service this op.